src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/CSharpRemoveUnreachableCodeDiagnosticAnalyzer.cs

```diff
@@ -109,39 +111,44 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
             // fix off of.
             var firstStatementLocation = root.SyntaxTree.GetLocation(firstUnreachableStatement.FullSpan);
 
-            if (!firstUnreachableStatement.IsParentKind(SyntaxKind.Block) &&
-                !firstUnreachableStatement.IsParentKind(SyntaxKind.SwitchSection))
-            {
-                // Can't actually remove this statement (it's an embedded statement in something 
-                // like an 'if-statement').  Just fade the code out, but don't offer to remove it.
-                if (fadeOutCode)
-                {
-                    context.ReportDiagnostic(
-                        Diagnostic.Create(UnnecessaryWithoutSuggestionDescriptor, firstStatementLocation));
-                }
-                return;
-            }
-
             // 'additionalLocations' is how we always pass along the locaiton of the first unreachable
             // statement in this group.
             var additionalLocations = SpecializedCollections.SingletonEnumerable(firstStatementLocation);
 
-            var descriptor = fadeOutCode ? UnnecessaryWithSuggestionDescriptor : Descriptor;
-
-            context.ReportDiagnostic(
-                Diagnostic.Create(descriptor, firstStatementLocation, additionalLocations));
+            if (fadeOutCode)
+            {
+                var tagIndices = ImmutableDictionary<string, IEnumerable<int>>.Empty
+                    .Add(nameof(WellKnownDiagnosticTags.Unnecessary), new int[] { 0 });
```

Comment:
Nit: for consistency and maintainability
```suggestion
                    .Add(nameof(WellKnownDiagnosticTags.Unnecessary), new int[] { additionalLocations.Length });
```

Reply:
Oh actually, you don't add to the additional locations below. Ignore this suggestion.